programs/bench.h

```diff
@@ -19,46 +19,130 @@ extern "C" {
 #define ZSTD_STATIC_LINKING_ONLY   /* ZSTD_compressionParameters */
 #include "zstd.h"     /* ZSTD_compressionParameters */
 
+typedef enum {
+    BMK_timeMode = 0,
+    BMK_iterMode = 1
+} BMK_loopMode_t;
+
+typedef enum {
+    BMK_both = 0,
+    BMK_decodeOnly = 1,
+    BMK_compressOnly = 2
+} BMK_mode_t;
+
+#define ERROR_STRUCT(baseType, typeName) typedef struct { \
+    int error;       \
+    baseType result; \
+} typeName
+
 typedef struct {
     size_t cSize;
     double cSpeed;   /* bytes / sec */
     double dSpeed;
 } BMK_result_t;
 
-/* 0 = no Error */
 typedef struct {
-	int errorCode;
-	BMK_result_t result;
-} BMK_return_t;
+    size_t size;    /* sum of return values */
+    U64 nanoSecPerRun; /* time per iteration */
+} BMK_customResult_t;
```

Comment:
From a naming perspective, `BMK_result_t` and `BMK_customResult_t` feel close.
But they are not.
They represent 2 very different levels of the API.
So it ends up being confusing.

Unfortunately, I'm short of any proposal at this stage.
So let's think about it a bit more.
Naming can be hard at times.